src/vm.rs

```diff
@@ -136,7 +71,59 @@ pub trait VirtualCPU {
 
 	fn args(&self) -> &[OsString];
 
-	fn cmdsize(&self, syssize: &mut SysCmdsize) {
+	/// `addr` is the address of the hypercall parameter in the guest's memory space. `data` is the
+	/// parameter that was send to that address by the guest.
+	fn address_to_hypercall(&self, addr: u16, data: usize) -> Option<Hypercall<'_>> {
+		if let Ok(hypercall_port) = HypercallAddress::try_from(addr) {
```

Comment:
Let's replace this `if let` with an `HypercallAddress::try_from(addr).ok()?` for an early return.

Reply:
Why does this return earlier?

Reply:
Because of the question mark operator :D